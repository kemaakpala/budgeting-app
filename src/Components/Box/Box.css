.box {
  padding:10px;
  background-color: #033649;
  width: 50%;
  border: 1px solid #031634;
  border-radius: 8px;
  margin-bottom: 1rem;
}