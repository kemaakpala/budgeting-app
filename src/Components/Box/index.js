import React from 'react'
import Divider from '../Divider'
import Typography from '../Typography'
import './Box.css'

export default (props) => {
  const {children, header, footer} = props;
  return (
    <div className="box">
      { header 
          ? <div className="box-header"><Typography component="h2">{header}</Typography></div>
          : null
      }
      <Divider />
      <div className="box-body">
        {children}
      </div>
      {
        footer
          ? <div className="box-footer">{footer}</div>
          : null
      }
    </div>
  )
}