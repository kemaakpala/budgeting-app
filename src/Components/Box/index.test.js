import React from 'react';
import { render } from '@testing-library/react';
import MockStoreProvider from '../../utils/mockStoreProvider';
import Box from './index';

const renderComponent = (props = {}) => (
  <MockStoreProvider>
    <Box {...props}>test</Box>
  </MockStoreProvider>
);

describe('<Box />', () => {
  it('it should render a snapshot with box body and className', () => {
    const props = {
      className: 'testClass'
    }
    const { container : { firstChild } } = render(renderComponent({ ...props }))
    expect(firstChild).toMatchSnapshot();
  });

  it('it should render a snapshot with box body, className, header', () => {
    const props = {
      className: 'testClass',
      header: 'testHeader'
    }
    const { container : { firstChild } } = render(renderComponent({ ...props }))
    expect(firstChild).toMatchSnapshot();
  });

  it('it should render a snapshot with box body, className, footer', () => {
    const props = {
      className: 'testClass',
      footer: 'testFooter'
    }
    const { container : { firstChild } } = render(renderComponent({ ...props }))
    expect(firstChild).toMatchSnapshot();
  });


  it('it should render a snapshot with box header, body, className, footer', () => {
    const props = {
      className: 'testClass',
      header: 'testHeader',
      footer: 'testFooter'
    }
    const { container : { firstChild } } = render(renderComponent({ ...props }))
    expect(firstChild).toMatchSnapshot();
  });
})