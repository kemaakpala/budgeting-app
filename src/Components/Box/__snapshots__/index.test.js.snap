// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> it should render a snapshot with box body and className 1`] = `
<div
  class="box testClass"
>
  <hr
    class="divider"
  />
  <div
    class="box-body"
  >
    test
  </div>
</div>
`;

exports[`<Box /> it should render a snapshot with box body, className, footer 1`] = `
<div
  class="box testClass"
>
  <hr
    class="divider"
  />
  <div
    class="box-body"
  >
    test
  </div>
  <div
    class="box-footer"
  >
    testFooter
  </div>
</div>
`;

exports[`<Box /> it should render a snapshot with box body, className, header 1`] = `
<div
  class="box testClass"
>
  <div
    class="box-header"
  >
    <h2>
       
      testHeader
       
    </h2>
     
  </div>
  <hr
    class="divider"
  />
  <div
    class="box-body"
  >
    test
  </div>
</div>
`;

exports[`<Box /> it should render a snapshot with box header, body, className, footer 1`] = `
<div
  class="box testClass"
>
  <div
    class="box-header"
  >
    <h2>
       
      testHeader
       
    </h2>
     
  </div>
  <hr
    class="divider"
  />
  <div
    class="box-body"
  >
    test
  </div>
  <div
    class="box-footer"
  >
    testFooter
  </div>
</div>
`;
